javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -2089,6 +2063,40 @@ public String toString() {
         }
     }
 
+    final class GroupedIterator<T> implements Iterator<Seq<T>> {
```

Comment:
I don't know if we should really switch to Iterator (instead of AbstractIterator) **in version 2.1.0** for slightly better performance...

In 3.0.0 I think we should use the faster Iterator (without a pretty toString()).